### PR TITLE
Fix AlertListener on IPv6-aware hosts and prevent YoutubeDL from making the plugin hang

### LINKED ISSUE #143: AlertListener does not work with some hosts on Linux
### Describe the Bug

Plex alert listener fails to start on Linux (when using ipv6?) and the following error appears in the logs.

`ERROR (alert:94) - getsockaddrarg: bad family`

### Expected Behavior

_No response_

### Additional Context

IPv6 is likely disabled in Plex's python build.

We can possibly patch python-plexapi to setup websocket-client properly. https://github.com/ReenigneArcher/python-plexapi/blob/c41d98170b3761b8edc7d75970a01b80bb6281c6/plexapi/alert.py#L55

https://websocket-client.readthedocs.io/en/v0.59.0/examples.html?highlight=socket#using-unix-domain-sockets

https://github.com/majek/puka/issues/27

https://github.com/qpython-android/qpysl4a/issues/2

### LINKED ISSUE #163: youtube-dl logs are going to system/container log
### Describe the Bug

Logs for youtube-dl are not being picked up in Themerr logs.

### Expected Behavior

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by zdimension**
## Description

- on many Linux machines, `localhost` resolves to `::1`. The `websocket` module used here expects an `AF_INET` address, so it expectedly crashes when passed an IPv6 address such as `::1`. The default URL used for reaching Plex is modified to `127.0.0.1` to prevent this.
- no code was present to handle YoutubeDL raising an exception. If such a case were to happen previously, the whole thread would crash; in practice this means that if any movie in the library can't be downloaded for any reason, the worker thread trying to download it will crash and stop processing any more content. In the worst case (e.g. mine) you can end up with all threads hung. Refreshing metadata or restarting the plugin doesn't change anything, the problem is logical and deterministic. Errors are now handled properly and printed to the plugin logs, and the process can continue.
- also, a typo in the YoutubeDL call is fixed (`outmpl` -> `outtmpl`)

### Issues Fixed or Closed
- Fixes #143
- Fixes #163

## Type of Change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [ ] Documentation update (changes to documentation)
- [ ] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I want maintainers to keep my branch updated

### CODE REVIEW

File: Contents/Code/youtube_dl_helper.py [hunk 1]
```diff
@@ -13,6 +13,26 @@
...
+    def error(self, msg, *args, **kwargs):
+        Log.Error(msg, *args, **kwargs)
+
+    def critical(self, msg, *args, **kwargs):
+        Log.Critical(msg, *args, **kwargs)
+
+    def exception(self, msg, *args, **kwargs):
+        Log.Exception(msg, exc_info=True, *args, **kwargs)
+
+youtube_logger = LoggerProxy()
```
Comment (ReenigneArcher) [inline Contents/Code/youtube_dl_helper.py:35]:
```suggestion
# get the plugin logger
plugin_logger = logging.getLogger(plugin_identifier)
```

File: Contents/Code/youtube_dl_helper.py [hunk 2]
```diff
@@ -13,6 +13,26 @@
 from typing import Optional
```
Comment (ReenigneArcher) [inline Contents/Code/youtube_dl_helper.py:13]:
```suggestion
from constants import plugin_identifier
from typing import Optional
```

File: Contents/Code/youtube_dl_helper.py [hunk 3]
```diff
@@ -35,7 +55,10 @@ def process_youtube(url):
     ...
     """
     youtube_dl_params = dict(
-        outmpl='%(id)s.%(ext)s',
+        logger=youtube_logger,
```
Comment (ReenigneArcher) [inline Contents/Code/youtube_dl_helper.py:58]:
```suggestion
        logger=plugin_logger,
```

File: Contents/Code/youtube_dl_helper.py [hunk 4]
```diff
@@ -35,7 +55,10 @@ def process_youtube(url):
...
+        verbose=True,
+        socket_timeout=10,
+        outtmpl=u'%(id)s.%(ext)s',
```
Comment (ReenigneArcher) [inline Contents/Code/youtube_dl_helper.py:61]:
```suggestion
```

Not sure what `outtmpl` is, seems like the output filename... but we don't actually download anything.

**Review by ReenigneArcher — CHANGES_REQUESTED** [review]
Thanks for the PR!

There's a simpler way to use the plugin logger.

Can you think of some way to force/test this exception situation?

File: Contents/Code/youtube_dl_helper.py (hunk 3 above)
Comment (ReenigneArcher) [reply Contents/Code/youtube_dl_helper.py:58]: Do you mind making all these parameters alphabetical?

File: Contents/Code/youtube_dl_helper.py (hunk 1 above)
Comment (zdimension) [reply Contents/Code/youtube_dl_helper.py:35]: Oh, well, that's certainly better. This is my first time writing code for Plex plugin-side so I wasn't sure if Plex has its own logging system or if I could use the regular `logging` module. I have a surefire way to trigger an exception; I'll just restart the plugin and let it try downloading the theme for one of the various movies with unavailable videos. I'll update

File: Contents/Code/youtube_dl_helper.py (hunk 3 above)
Comment (zdimension) [reply Contents/Code/youtube_dl_helper.py:58]: Oops, I added all of those in nano and didn't run a formatter afterwards. I'll fix that

File: Contents/Code/youtube_dl_helper.py [hunk 5]
```diff
@@ -10,9 +10,13 @@
     from plexhints.prefs_kit import Prefs  # prefs kit
 
 # imports from Libraries\Shared
+from constants import plugin_identifier
 from typing import Optional
+import logging
```
Comment (ReenigneArcher) [inline Contents/Code/youtube_dl_helper.py:15]:
```suggestion
```

Can you throw this at the top?

```diff
# -*- coding: utf-8 -*-

+# standard imports
+import logging

# plex debugging
```

File: Contents/Code/plex_api_helper.py [hunk 6]
```diff
@@ -38,7 +38,7 @@
 q = queue.Queue()
 
 
-plex_url = 'http://localhost:32400'
+plex_url = 'http://127.0.0.1:32400'  # the explicit IPv4 address is used because `localhost` can resolve to ::1, which `websocket` rejects
```
Comment (ReenigneArcher) [inline Contents/Code/plex_api_helper.py:41]: line too long, 120 is the limit

I kind of think it would be cleaner to separate this to a separate PR as well.
